### PR TITLE
🎉 New Source: Audience Project

### PR DESCRIPTION
**Pull request by Zawar92**
## What
Building a new connector [Audience Project](https://www.audienceproject.com/resources/api-documentation/). As related to the current [issue](https://github.com/airbytehq/airbyte/issues/26424)
## How
*Describe the solution*

## Recommended reading order
1. `x.java`
2. `y.python`

## 🚨 User Impact 🚨
*Are there any breaking changes? What is the end result perceived by the user?*

*For connector PRs, use this section to explain which type of semantic versioning bump occurs as a result of the changes. Refer to our [Semantic Versioning for Connectors](https://docs.airbyte.com/contributing-to-airbyte/#semantic-versioning-for-connectors) guidelines for more information. **Breaking changes to connectors must be documented by an Airbyte engineer (PR author, or reviewer for community PRs) by using the [Breaking Change Release Playbook](https://docs.google.com/document/d/1VYQggHbL_PN0dDDu7rCyzBLGRtX-R3cpwXaY8QxEgzw/edit).***

*If there are breaking changes, please merge this PR with the 🚨🚨 emoji so changelog authors can further highlight this if needed.*

## Pre-merge Actions
*Expand the relevant checklist and delete the others.*

<details><summary><strong>New Connector</strong></summary>

### Community member or Airbyter

- **Community member?** Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added and passing. Community members, please provide proof of success locally e.g: screenshot or copy-paste unit, integration, and acceptance test output. To run acceptance tests for a Python connector, follow instructions in the README. For java connectors run `./gradlew :airbyte-integrations:connectors:<name>:integrationTest`.
- Connector version is set to `0.0.1`
    - `Dockerfile` has version `0.0.1`
- Documentation updated
    - Connector's `README.md`
    - Connector's `bootstrap.md`. See [description and examples](https://docs.google.com/document/d/1ypdgmwmEHWv-TrO4_YOQ7pAJGVrMp5BOkEVh831N260/edit?usp=sharing)
    - `docs/integrations/<source or destination>/<name>.md` including changelog with an entry for the initial version. See changelog [example](https://docs.airbyte.io/integrations/sources/stripe#changelog)
    - `docs/integrations/README.md`

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Updating a connector</strong></summary>

### Community member or Airbyter

- Grant edit access to maintainers ([instructions](https://docs.github.com/en/github/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork#enabling-repository-maintainer-permissions-on-existing-pull-requests))
- Unit & integration tests added

### Airbyter

If this is a community PR, the Airbyte engineer reviewing this PR is responsible for the below items.

- Create a non-forked branch based on this PR and test the below items on it
- Build is successful
- If new credentials are required for use in CI, add them to GSM. [Instructions](https://docs.airbyte.io/connector-development#using-credentials-in-ci).

</details>

<details><summary><strong>Connector Generator</strong></summary>

- Issue acceptance criteria met
- PR name follows [PR naming conventions](https://docs.airbyte.com/contributing-to-airbyte/issues-and-pull-requests#pull-request-title-convention)
- If adding a new generator, add it to the [list of scaffold modules being tested](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connector-templates/generator/build.gradle#L41)
- The generator test modules (all connectors with `-scaffold` in their name) have been updated with the latest scaffold by running `./gradlew :airbyte-integrations:connector-templates:generator:testScaffoldTemplates` then checking in your changes
- Documentation which references the generator is updated as needed

</details>

### CODE REVIEW

**Comment by Zawar92** [comment]
Test Results:
1. Integration test passes.
integration_test


2. Gradle Python Format passes.
Gradel PythonFormat


3. Gradle integration test failing due to timeout. Asked questions on the [discussion on airbyte](https://discuss.airbyte.io/t/timeout-issue-in-local-gradlew-airbyte-integrations-name-integrationtest/4625).

Gradel Integration test

File: airbyte-integrations/connectors/source-audienceproject/source_audienceproject/streams_report.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+            parent_records = parent.read_records(
+                sync_mode=sync_mode, cursor_field=cursor_field, stream_slice=stream_slice, stream_state=stream_state
+            )
+            for record in parent_records:
+                yield {"campaign_id": record.get("id")}
+
+    def path(
+        self, stream_state: Mapping[str, Any] = None, stream_slice: Mapping[str, Any] = None, next_page_token: Mapping[str, Any] = None
+    ) -> str:
+        return f"reports/{stream_slice['campaign_id']}"
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-audienceproject/source_audienceproject/streams_report.py:48]: A lot of streams are duplicating code. Please create a base class where you can just create the stream using the code from the base class. Usually connectors keep the streams in a single file.

File: airbyte-integrations/connectors/source-audienceproject/source_audienceproject/source.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+        auth = AudienceProjectAuthenticator(url_base=AudienceprojectStream.oauth_url_base, config=config)
+        auth_header = auth.get_auth_header()
+        url_base = AudienceprojectStream.url_base + "campaigns"
+        try:
+            response = requests.get(url=url_base, headers=auth_header)
+            response.raise_for_status()
+            return True, None
+        except Exception as e:
+            return False, e
```
Comment (marcosmarxm) [inline airbyte-integrations/connectors/source-audienceproject/source_audienceproject/source.py:31]: Try to create a sample stream and read the first page. For more info check https://discuss.airbyte.io/t/how-the-airbyte-cdk-check-command-works/4589

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

**Comment by marcosmarxm** [comment]
Hello @Zawar92  thanks so much for your effort but the project is not accepting Python CDK sources anymore. This is a decision to improve maintenance and support. You can use the version for your project and if you want to contribute we recommend to move the connector to low-code/no-code framework.

Hope you understand the decision,
Bests
Marcos

**Comment by Zawar92** [comment]
> Hello @Zawar92 thanks so much for your effort but the project is not accepting Python CDK sources anymore. This is a decision to improve maintenance and support. You can use the version for your project and if you want to contribute we recommend to move the connector to low-code/no-code framework.
> 
> Hope you understand the decision, Bests Marcos

Hi @marcosmarxm,

I can understand your decision. Can you give me some understanding of how we can convert this to low code? For low-code, It would require my side to add some Python code for specific methods like `next_page_token` and `stream_slices`. It would be a good hint for me and I will add a new PR with the low code for the connector. 
Thank you.

**Comment by marcosmarxm** [comment]
@Zawar92 I'm more than glad to help you migrating this to low-code. Send me a message on Slack. Right now we're most dedicated to reviewing August Hackathon but can provide you some tips to make the transition!

**Pull request closed without merging**